src/main/java/pikabot/Storage.java

```diff
@@ -0,0 +1,136 @@
+     * @param f Data file to be read from.
+     * @return TaskList of tasks parsed from data file.
+     * @throws FileNotFoundException If file is not found.
+     */
+    public static TaskList fileToTaskList(File f) throws FileNotFoundException {
+
+        Scanner sc = new Scanner(f);
+        ArrayList<Task> arrList = new ArrayList<>();
+        TaskList taskList = new TaskList(arrList);
+
+        while (sc.hasNextLine()) {
+            String taskStr = sc.nextLine();
+            char taskType = taskStr.charAt(1);
+            boolean isTaskDone = (taskStr.charAt(4) == 'X');
+            taskStr = taskStr.substring(7);
+
+            if (taskType == 'D') {
+                String[] taskDetails = taskStr.split("\\(by: ", 2);
+                String description = taskDetails[0];
+                String date = taskDetails[1].substring(0, taskDetails[1].length()-1);
+                DateTimeFormatter dtf = DateTimeFormatter.ofPattern("d-MMM-yyyy");
+                Deadline deadline = new Deadline(description, LocalDate.parse(date, dtf));
+                if (isTaskDone) {
+                    deadline.markAsDone();
+                }
+                taskList.add(deadline);
+
+            } else if (taskType == 'E') {
+                String[] taskDetails = taskStr.split("\\(at: ", 2);
+                String description = taskDetails[0];
+                String date = taskDetails[1].substring(0, taskDetails[1].length()-1);
+                DateTimeFormatter dtf = DateTimeFormatter.ofPattern("d-MMM-yyyy");
+                Event event = new Event(description, LocalDate.parse(date, dtf));
+                if (isTaskDone) {
+                    event.markAsDone();
+                }
+                taskList.add(event);
+
+            } else {
+                Todo todo = new Todo(taskStr);
+                if (isTaskDone) {
+                    todo.markAsDone();
+                }
+                taskList.add(todo);
+            }
+        }
+        return taskList;
+    }
+
+    /**
+     * Writes tasks to data file on computer.
+     *
+     * @param taskList List of tasks.
+     * @throws IOException If the named file exists
+     * but is a directory rather than a regular file,
+     * does not exist but cannot be created, or cannot
+     * be opened for any other reason.
+     */
+    public void TaskListToFile(TaskList taskList) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task task : taskList.getTaskList()) {
+            fw.write(task.toString() + "\n");
+        }
+        fw.close();
+    }
+
+    /**
+     * Adds new task to data file on computer.
+     *
+     * @param taskToAdd Task to be added to data file on computer.
+     * @throws IOException If the named file exists but is
+     * a directory rather than a regular file, does not exist
+     * but cannot be created, or cannot be opened for any other reason.
+     */
+    public void appendToFile(Task taskToAdd) throws IOException{
```

Comment:
I like how clean this bit of code is! But, with regards to the coding standard, there should be a space after <code> IOException </code>, other than that good job ! 😄